### PR TITLE
Fix first hitobject in osu! hidden mod not getting correct fade applied

### LINKED ISSUE #8153: First hitcircle not vanishing with HD
When you play Hidden on osu! stable the approach circle of the first object (unless changed in the settings) shows up while the object itself vanishes like all others.
In osu!lazer the object stays visible however.
This could lead to some weird reading situations if the beatmap starts with a stream or burst.

Screenshots:
https://imgur.com/a/Yg8nREW
The first screenshot is what the beginning of the beatmap looks like nomod.
The second one shows how it looks with HD on osu!lazer.
The third one shows how it would look on stable.

osu!lazer version:
2020.306.0

Logs:
[runtime.log](https://github.com/ppy/osu/files/4298136/runtime.log)
[updater.log](https://github.com/ppy/osu/files/4298137/updater.log)
[database.log](https://github.com/ppy/osu/files/4298138/database.log)
[network.log](https://github.com/ppy/osu/files/4298139/network.log)
[performance.log](https://github.com/ppy/osu/files/4298140/performance.log)

### PR DESCRIPTION
**Pull request by peppy**
Closes #8153

### CODE REVIEW

File: osu.Game/Rulesets/Mods/ModHidden.cs [hunk 1]
```diff
@@ -38,7 +38,13 @@ public void ReadFromConfig(OsuConfigManager config)
         public virtual void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables)
         {
             if (IncreaseFirstObjectVisibility.Value)
+            {
+                var firstObject = drawables.FirstOrDefault();
```
Comment (bdach) [inline osu.Game/Rulesets/Mods/ModHidden.cs:42]: I believe this should be

```suggestion
                var firstObject = drawables.FirstOrDefault(IsFirstHideableObject);
```

or, probably better, rewrite this logic a bit to be

```csharp
if (IncreaseFirstObjectVisibility.Value)
{
    drawables = drawables.SkipWhile(h => !IsFirstHideableObject(h));

    var firstObject = drawables.FirstOrDefault();
    if (firstObject != null)
        firstObject.ApplyCustomUpdateState += ApplyFirstObjectIncreaseVisibilityState;

    drawables = drawables.Skip(1);
}
```

Can be tested on any map that starts with a spinner - the second object won't fade out at all while the third will work like normal hidden.

**Review by bdach — CHANGES_REQUESTED** [review]

**Review by bdach — DISMISSED** [review]

**Comment by smoogipoo** [comment]
The weird part is that the approach circle is not supposed to fade out, but I'm not good enough at HD to know how confusing this actually is. Maybe this should instead fade out just `circle.CirclePiece` if `increasedVisibility = true`?

**Comment by bdach** [comment]
That's what `master` does already, doesn't it?

**Comment by smoogipoo** [comment]
Master:
- Doesn't fade out anything.

This PR:
- Fades out everything.

Stable:
- Fades out only the circle.

**Comment by bdach** [comment]
Oh the issue is formulated weirdly. Sure.

**Comment by peppy** [comment]
That wasn't the intention.

**Comment by smoogipoo** [comment]
To me it feels weird because the approach is hidden way before it touches the circle, making it kinda useless? Again I can't really provide much insight here but the reason for it not hiding in the first place is because players maintain an afterimage of the circle in their mind specifically so they can hit the first note accurately, assuming that the approach is visible.

**Comment by peppy** [comment]
Okay, will change it to match stable for now.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**